contracts/WpxCvx.sol

```diff
@@ -0,0 +1,223 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.12;
+
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+import {ReentrancyGuard} from "@rari-capital/solmate/src/utils/ReentrancyGuard.sol";
+import {ERC20} from "@rari-capital/solmate/src/tokens/ERC20.sol";
+import {SafeTransferLib} from "@rari-capital/solmate/src/utils/SafeTransferLib.sol";
+import {PirexCvx} from "./PirexCvx.sol";
+import {ICurvePool} from "./interfaces/ICurvePool.sol";
+
+contract WpxCvx is ERC20, Ownable, ReentrancyGuard {
+    using SafeTransferLib for ERC20;
+
+    // Enumeration for the token swap
+    enum Token {
+        CVX,
+        pxCVX
+    }
+
+    ERC20 public immutable pxCVX;
+    ERC20 public immutable CVX;
+
+    // CVX token index used for identification purposes in the curvePool
+    uint256 public cvxIndex;
```

Comment:
I don't think this needs to be stored. Passing in the token indexes to the `swap` call also reduces gas costs by 0.5%.